Khan/antwerp
# implement tool for publishing packages from a mono-repo

The reason for creating this tool is that `lerna` does a couple of things things which we'd like to avoid with publishing packages from the wonder-blocks mono-repo:
- making changes to a dependency will cause `lerna` to automatically bump the version of packages that depend on it… this feels fundamentally wrong since the dependee may not be ready to use the updated version of the dependency.  Also, if the dependee is using version ^1.0.0 and the dependency is bumped to 1.1.0, then there's no reason the update the dependee's version. 
- right now `lerna` is responsible for bumping versions (as well as publishing) which means that you have to commit changes without bumping versions.  In repos that publish a single package this is fine b/c other packages can’t rely on in between states.  This is different in mono-repos b/c you may change the version of a dependency in the same commit that you change a package to rely on that new version.  This means that we should be bumping versions in the same commit that we’re making this change.  This avoids having commits in the repo that you can never checkout and run b/c of an inconsistent state.  The publish (and tagging) can and should be done in a separate commit. 

TODO:
- [x] set up travis-ci
- [x] push the created tags to GitHub after publishing

Test Plan:
- `yarn test`
- `npm link`
- checkout https://github.com/kevinbarabash/workspaces-test 
- bump versions of `foo` and `bar` packages in `workspaces-test` and commit changes
- run `antwerp --dry-run` from `workspaces-test` folder
- see that the output contains commands to tag the most recent commit with `@kevinbarabash/foo@0.1.10` and `@kevinbarabash/bar@0.1.11` and publish `@kevinbarabash/foo` and `@kevinbarabash/bar` packages.  Notice that it doesn't try to bump or publish `@kevinbarabash/foobar` which relies on version `^0.1.0` of both of those packages.
- run `antwerp` again without the `--dry-run` flag and verify that it's published those packages to https://npmjs.org/~kevinbarabash and that the most recent commit has indeed been tagged appropriately.
